data/campaigns/fri05.wmf/scripting/init_p1.lua

```diff
@@ -0,0 +1,2975 @@
+          x= 80,
+          y= 169,
+          wares=             {
+            },
+        },
+      [124] =         {
+          x= 83,
+          y= 173,
+          wares=             {
+            },
+        },
+    },
+}
+
+function place_flags(eco)
+
+   for i = 1, #eco.flags do
+      local flag = p1:place_flag(map:get_field(eco.flags[i].x, eco.flags[i].y),true)
+      for ware,count in pairs(eco.flags[i].wares) do
+         flag:set_wares(ware,count)
+      end
+   end
+end
+
+function place_roads(eco)
+   for i = 1, #eco.roads do
+      local command = string.format("road = p1:place_road(\"normal\", map:get_field(%i,%i).immovable, %s ,true)", eco.roads[i].x, eco.roads[i].y, eco.roads[i].dirs)
+      local f = assert (load (command))
+      f()
+      -- Second carrier can't be placed currently, so we just fake a carrier
+      worker = (p1.tribe_name.."_carrier")
+      road:set_workers(worker,1)
+   end
+end
+
+function place_buildings(eco)
+
+   for i = 1, #eco.buildings do
+      building = p1:place_building(eco.buildings[i].name, map:get_field(eco.buildings[i].x, eco.buildings[i].y), false, true)
+
+      if eco.buildings[i].inputs then
+         for ware,count in pairs(eco.buildings[i].inputs) do
+            building:set_inputs(ware,count)
+         end
+      end
+
+      if eco.buildings[i].workers then
+         local up = {}
+         for worker,count in pairs(eco.buildings[i].workers) do
+            if count == 0 then
+               up[#up+1] = worker
+            elseif count > 1 and #up > 0 and not building.descr.type_name == "warehouse" then
+               for i = 1, count - 1 do
+                  building:set_workers(up[i],1)
+               end
+               building:set_workers(worker,1)
+            else
+               building:set_workers(worker,1)
+            end
+         end
+      end
+
+      if eco.buildings[i].wares then
+         for ware,count in pairs(eco.buildings[i].wares) do
+            building:set_wares(ware,count)
+         end
+      end
+
+      if eco.buildings[i].soldiers then
+         for soldier,count in pairs(eco.buildings[i].soldiers) do
+            building:set_soldiers({soldier:sub(1,1),soldier:sub(2,2),soldier:sub(3,3),soldier:sub(4,4)}, count)
+         end
+      end
+   end
+
+end
+
+place_flags(eco)
+place_roads(eco)
+place_buildings(eco)
+
+-- End of autogenerated section
+
+local soldiers = {}
+local max_takeover_soldiers = {300, 200, 100}
+local total_soldiers = 0
+for descr,n in pairs(campaign_data) do
+   if descr ~= "000" and n > 0 then
+      total_soldiers = total_soldiers + n
+   end
+end
+if total_soldiers > max_takeover_soldiers[difficulty] then total_soldiers = max_takeover_soldiers[difficulty] end
+while total_soldiers > 0 do
+   local key1 = math.random(0, 2)
+   local key2 = math.random(0, 6)
+   local key3 = math.random(0, 2)
+   local descr = key1 .. key2 .. key3
+   if not (key1 == 0 and key2 == 0 and key3 == 0) and campaign_data[descr] ~= nil then
```

Comment:
this means the player is betrayed of the number of rookies he brought over from the last scenario.

Reply:
No, we first carry over only trained soldiers, and if the maximum is not reached yet then we also carry over rookies below. This is to ensure that trained soldiers are preferred in carryover.